### ISSUE TITLE
More bugs in limit()

### ISSUE THREAD

**Issue by akazakov** [RoaringBitmap/RoaringBitmap#116]
Here's code that triggers it:

```
    public static void limitBug2() {
        class MyConsumer implements IntConsumer {
            public int count = 0;
            @Override public void accept(int value) { count++; }
        }

        RoaringBitmap r = new RoaringBitmap();
        int count = 0;
        for (int i = 0; i < 500; i++) {
            for (int j = 0; j < 9943; j++) {
                if (i % 2 == 0) r.add(count); count++;
            }
        }
        System.out.println("Source cardinality:  " + r.getCardinality());

        RoaringBitmap limited = r.limit(1000000);
        System.out.println("Limited cardinality:  " + limited.getCardinality());
        MyConsumer c = new MyConsumer();

        limited.forEach(c);
        System.out.println("Iteration count: " + c.count);  // <-- This should be 1000000
        System.out.println(r.toArray().length);
        System.out.println(limited.toArray().length);         // <-- This will blow up!
    }
```

Here's the output 

```
Source cardinality:  2485750
Limited cardinality:  1000000
Iteration count: 1000004
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 1000000
	at org.roaringbitmap.BitmapContainer.fillLeastSignificant16bits(BitmapContainer.java:355)
	...
```

**Comment by lemire** [comment]
Thanks. We will check it out soon.

**Comment by lemire** [comment]
Fixed in release 0.6.21.

**Comment by lemire** [comment]
Please keep the bug reports coming!
